vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -113,24 +113,3 @@ def get_parent_tags_recursively(mor, infrastructure_data):
         parent_tags.extend(tags)
         return parent_tags
     return []
-
-
-def should_collect_per_instance_values(metric_name, resource_type):
-    # TODO: Implement. For now we don't collect per-instance level metrics (aka per-core for cpu, per-vm for disk etc.)
-    # TODO: Collecting per-instance metrics is really expensive for big environments and has usually little value.
-    # TODO: Also that adds an extra layer of complexity where users have to set `instance:none` to see the correct
-    # value.
-    return False
-
-
-def get_mapped_instance_tag(metric_name):
```

Comment:
I think that it is a much better user experience to replace `metric_instance` with the meaning of the instance tag (`cpu_core`, `vm_id`, etc.)

Reply:
Just saw that part in the PR description:

> - simpler to implement, read, maintain
> - it's consistent with the vSphere documentation: https://code.vmware.com/apis/704/vsphere/vim.PerformanceManager.MetricId.html
> - using different tag keys might lead to confusions and inconsistencies since we need to arbitrarily define them and also maintain them.

1. Agree that makes the code simpler
2. It's consistent because this is an actual limitation on vSphere metrics storage side. From the user experience side, it is much easier to read a meaningful tag key that `metric_instance`
3. I actually think that using different tag keys will help users a lot to avoid confusions, the name of the tag will be enough to understand what it is. Yes we need to arbitrarily define them and maintain them (if they change in the future), but we can always default back to `instance` or `metric_instance` when we don't have an existing mapping.

Reply:
That's fair. Let's use specific specific tags when possible for instance metrics then.

Another option is to provide both: tagged by `instance` and specific tag `vm_id, cpu_core` with the same value.